### DISCUSSION TITLE
App Store Connect upload: Asset validation failed due to Invalid sdk value (18.2 > 18.1)

### DISCUSSION THREAD

**Discussion by mikeymike9000** [firebase/firebase-ios-sdk#14148] [Q&A] [answered]
Xcode 16.2 beta 2, macOS 15.2 beta 3

When trying to upload my app to App Store Connect it fails with:

> Asset validation
> Invalid sdk value. The value provided for the sdk portion of LC_BUILD_VERSION in
> TW.app/ > Frameworks/FirebaseAnalytics.framework/FirebaseAnalytics is 18.2 which
> is greater than the maximum allowed value of 18.1.

Is this on Apple's side because App Store Connect doesn't yet accept SDK 18.2, i.e. do I just need to wait for Apple to accept 18.2 or else I could install an older version of Xcode?

Or is there something I can do on my end so Firebase Frameworks are 18.1?

Or is this something to do with Firebase that will be 'fixed' in the next release?

![fail](https://github.com/user-attachments/assets/66e7102f-ea15-4bd9-89cc-4e842f588d52)

**Comment by ncooke3** [comment, marked as answer]
Yes, I believe this error is due to attempting to deploy with a beta Xcode. I would recommend trying with Xcode 16.1 as that is the newest, non-beta version.

**Reply by mikeymike9000** [reply]
ok thanks I was hoping it is an Apple 'issue'.

**Reply by mikeymike9000** [reply]
Confirmed.. I installed Xcode 16.1 (with its built-in iOS 18.1 SDK, not 18.2) and now App Store Connect accepts the Xcode 16.1 upload to TestFlight. There's no issue having both Xcode versions installed (when switching back and forth I execute xcode-select for command-line tools, delete Derived Data, rebuild dependencies).

**Reply by ncooke3** [reply]
Thank you for the follow-up!
